cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/ghodss/yaml"
+
+	clientset "k8s.io/client-go/kubernetes"
+	kubeadmapiext "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha1"
+	"k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade"
+	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
+)
+
+// upgradeVariables holds variables needed for performing an upgrade or planning to do so
+type upgradeVariables struct {
+	client        clientset.Interface
+	cfg           *kubeadmapiext.MasterConfiguration
+	versionGetter upgrade.VersionGetter
+}
+
+// EnforceRequirements verifies that it's okay to upgrade and then returns the variables needed for the rest of the procedure
+func EnforceRequirements(kubeConfigPath, cfgPath string, printConfig, nonInteractively bool) (*upgradeVariables, error) {
+	client, err := kubeconfigutil.ClientSetFromFile(kubeConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't create a Kubernetes client from file %q: %v", kubeConfigPath, err)
+	}
+
+	// Run healthchecks against the cluster
+	if err := upgrade.VerifyClusterHealth(client); err != nil {
+		return nil, err
+	}
+
+	// Fetch the configuration from a file or ConfigMap and validate it
+	cfg, err := upgrade.FetchConfiguration(client, os.Stdout, cfgPath, nonInteractively)
```

Comment:
I had it to be; kubeadm will ask the user if it should proceed with the default config if it didn't exist in-cluster. Now we can require every v1.7 user to do `kubeadm config upload from-flags`